libvast_test/vast/test/type_ids.hpp

```diff
@@ -0,0 +1,31 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/view.hpp"
+
+#include <caf/type_id.hpp>
+
+#include <string_view>
+
+namespace vast {
+template <class T>
+class container_view_handle;
+}
+
+CAF_BEGIN_TYPE_ID_BLOCK(vast_ut_block, 50000)
+  CAF_ADD_TYPE_ID(vast_ut_block, (vast::pattern_view))
+  CAF_ADD_TYPE_ID(vast_ut_block,
+                  (vast::container_view_handle<vast::list_view_ptr>))
+  CAF_ADD_TYPE_ID(vast_ut_block,
+                  (vast::container_view_handle<vast::map_view_ptr>))
+  CAF_ADD_TYPE_ID(vast_ut_block,
+                  (vast::container_view_handle<vast::record_view_ptr>))
+  CAF_ADD_TYPE_ID(vast_ut_block, (std::string_view))
+CAF_END_TYPE_ID_BLOCK(vast_ut_block)
```

Comment:
I don't think you need this at all if you wrap the `view<{list,map,record}>` objects into a `data_view` with `make_data_view(...)`.

Generally, view's should not be inspectable as they cannot be deserialized into by design.

Reply:
I think this is triggered in CHECK_EQUAL(data_view, data_view)

data_view is a caf::variant and it requires all types to be inspectable in inspect(auto& inspector, caf::variant);

I could replace the mentioned code by replacing comparisons with data_view into materialize(data_view)
It should work, right?

Reply:
Yes, that would work. It's probably safer to do that way.